tests/scripts/analyze_outcomes.py

```diff
@@ -95,32 +95,50 @@ def analyze_coverage(results, outcomes, allow_list, full_coverage):
             else:
                 results.warning('Allow listed test case was executed: {}', key)
 
+def name_matches_pattern(name, str_or_re):
+    """Check if name matches a pattern, that may be a string or regex.
+    - If the pattern is a string, name must be equal to match.
+    - If the pattern is a regex, name must fully match.
+    """
+    # The CI's python is too old for re.Pattern
+    #if isinstance(str_or_re, re.Pattern):
+    if not isinstance(str_or_re, str):
+        if str_or_re.fullmatch(name):
+            return True
+    else:
+        if str_or_re == name:
+            return True
+    return False
```

Comment:
Debatable: I find this more complex than
```suggestion
    if not isinstance(str_or_re, str):
        return str_or_re.fullmatch(name)
    else:
        return str_or_re == name:
```

Reply:
I tend to agree.